api/users/serializers.py

```diff
@@ -381,6 +389,17 @@ def get_two_factor_enabled(self, obj):
         except TwoFactorUserSettings.DoesNotExist:
             return False
 
+    def get_two_factor_confirmed(self, obj):
+        two_factor_addon = obj.get_addon('twofactor')
```

Comment:
Done! Had a bit of trouble with the implementation for this guy, so figured might as well outline what I tried:

For node settings, the `wiki_addon` is added to the context just for the `UpdateSerializer`. Doing this for the `UserSettingsSerializer` and adding it in the context for `UserSettingsUpdateSerializer` caused the `twofactor_addon` key to not be available in the context for `get` requests when accessing the `get_two_factor_enabled` and `get_two_factor_confirmed` properties. I tried adding it to the UserSettingsUpdateSerializer as well for use in the `update` method, but had no luck accessing it (as update is called before to_representation). Figured I'd just do it this way instead of banging my head against it any more, and see if anyone had ideas in CR!

Reply:
@erinspace - I agree with your caching of the `twofactor_addon` on `UserSettingsSerializer.to_representation` It's a good happy medium for reducing # of requests on both GET and PATCH requests.  In some places, we can't cache the two-factor addon ahead of time anyway, because our request creates it.

Also, caching there, instead of UserSettings.get_serializer_context requires less time and queries since the User object is already loaded.